src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +378,313 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
         }
 
+        private void GetRegionStart(Region v2Region, Uri uri, out int startLine, out int startColumn)
+        {
+            startLine = 0;
+            startColumn = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                // Read charOffset characters
+                char[] buffer = new char[v2Region.CharOffset];
+                reader.Read(buffer, 0, buffer.Length);
+
+                string text = new string(buffer);
+                string[] parts = text.Split(new string[] { "\r\n" }, StringSplitOptions.None);
+                startLine = parts.Length;
+                startColumn = parts.Last().Length + 1;
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+            }
+        }
+
+        private int GetRegionByteOffset(Region v2Region, Uri uri)
+        {
+            int result = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                if (reader != null)
+                {
+                    char[] buffer = new char[v2Region.CharOffset];
+
+                    // Read everything up to charOffset
+                    if (reader.ReadBlock(buffer, 0, buffer.Length) > 0)
+                    {
+                        result = encoding.GetByteCount(buffer, 0, buffer.Length);
+                    }
+                }
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+            }
+
+            return result;
+        }
+
+        private int GetRegionByteLength(Region v2Region, Uri uri)
+        {
+            int result = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                if (v2Region.StartLine > 0) // Use line and column 
+                {                    
+                    string sourceLine = null;
+
+                    // Read down to the line before startLine
+                    for (int i = 1; i < v2Region.StartLine; sourceLine = reader.ReadLine(), i++) { }
+
+                    // Read to startColumn
+                    char[] buffer = new char[v2Region.StartColumn];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    // Read the next charLength characters
+                    buffer = new char[v2Region.CharLength];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    result = encoding.GetByteCount(buffer);
+                }
+                else // Use charOffset
+                {
```

Comment:
You have to read (skip) the the first `charOffset` characters, and then read the next `charLength` characters. As the code stands, you're just reading  the first `charOffset` characters.

Reply:
Whoops. Fixed